packages/authentication/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/authentication/FirebaseAuthenticationPlugin.java

```diff
@@ -840,6 +840,36 @@ public void updateEmail(PluginCall call) {
         }
     }
 
+    @PluginMethod
+    public void verifyBeforeUpdateEmail(PluginCall call) {
+        try {
+            String newEmail = call.getString("newEmail");
+            if (newEmail == null) {
+                call.reject(ERROR_NEW_EMAIL_MISSING);
+                return;
+            }
+
+            FirebaseUser user = implementation.getCurrentUser();
+            if (user == null) {
+                call.reject(ERROR_NO_USER_SIGNED_IN);
+                return;
+            }
+
+            JSObject settings = call.getObject("actionCodeSettings");
+            if (settings == null) {
+                call.reject(ERROR_ACTION_CODE_SETTINGS_MISSING);
+                return;
+            }
+            ActionCodeSettings actionCodeSettings = FirebaseAuthenticationHelper.createActionCodeSettings(settings);
+            
+            implementation.verifyBeforeUpdateEmail(user, newEmail, actionCodeSettings, () -> call.resolve());
```

Comment:
You should also deal with the errors here (for example see `confirmVerificationCode`).